### PR TITLE
Feature Collection Measurements

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Feature Columns sollten auch ein Mesasurement haben](https://trello.com/c/M0tAveC8/134-feature-columns-sollten-auch-ein-mesasurement-haben)

File: operators/src/engine/result_descriptor.rs [hunk 1]
```diff
@@ -95,22 +95,37 @@ impl RasterResultDescriptor {
...
+        F: Fn(&HashMap<String, VectorColumnInfo>) -> HashMap<String, VectorColumnInfo>,
     {
         Self {
             data_type: self.data_type,
             spatial_reference: self.spatial_reference,
             columns: f(&self.columns),
         }
     }
+
+    pub fn column_data_type(&self, column: &str) -> Option<FeatureDataType> {
```
Comment (michaelmattig) [inline operators/src/engine/result_descriptor.rs:122]: shouldn't this return a `Result` and give an columnNotFound error?

File: operators/src/mock/mock_feature_collection_source.rs [hunk 2]
```diff
@@ -77,6 +79,7 @@ where
 {
     pub collections: Vec<FeatureCollection<G>>,
     pub spatial_reference: SpatialReferenceOption,
+    measurements: HashMap<String, Measurement>,
```
Comment (michaelmattig) [inline operators/src/mock/mock_feature_collection_source.rs:82]: maybe make them optional

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Changed
+
+- Added `Measurement`s to vector data workflows
+  - https://github.com/geo-engine/geoengine/pull/557
+  - The `VectorResultDescriptor`'s field `columns` now has two fields: `data_type` and `measurement`.
```
Comment (michaelmattig) [inline CHANGELOG.md:14]: add that the measurements are tracked by the operators automatically

File: operators/src/engine/result_descriptor.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply operators/src/engine/result_descriptor.rs:122]: But a HashMap also gives an Option back and not an error that states: not found.
So… there is only one meaning currently :smile:

File: operators/src/mock/mock_feature_collection_source.rs (hunk 2 above)
Comment (ChristianBeilschmidt) [reply operators/src/mock/mock_feature_collection_source.rs]: [f8f1487](https://github.com/geo-engine/geoengine/pull/557/commits/f8f148775eb85f563123802a11d7d0ad436cae8e)

File: CHANGELOG.md (hunk 3 above)
Comment (ChristianBeilschmidt) [reply CHANGELOG.md:14]: [e29fa8b](https://github.com/geo-engine/geoengine/pull/557/commits/e29fa8b4f7f2f937373fed14e26c1bea625e5842)

File: operators/src/engine/result_descriptor.rs (hunk 1 above)
Comment (michaelmattig) [reply operators/src/engine/result_descriptor.rs:122]: yeah that's exactly the reason why we do not use hashmaps directly but wrap them and handle the cases where something is not where it is supposed t obe.

Comment (ChristianBeilschmidt) [reply operators/src/engine/result_descriptor.rs:122]: But this is not a user API. You have to wrap it into an error anyway at your, for instance, operator. This is a convenience function for accessing the hash map with more than one value.

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**